docs/ceph-csi-upgrade.md

```diff
@@ -65,6 +66,38 @@ controller and snapshot CRD. more info can be found
 Refer [upgrade-from-v3.2-v3.3](https://github.com/ceph/ceph-csi/blob/v3.3.1/docs/ceph-csi-upgrade.md)
 to upgrade from cephcsi v3.2 to v3.3
 
+## Upgrading from v3.4 to v3.5
+
+**Ceph-csi releases from devel are expressly unsupported.** It is strongly
+recommended that you use [official
+releases](https://github.com/ceph/ceph-csi/releases) of Ceph-csi. Unreleased
+versions from the devel branch are subject to changes and incompatibilities
+that will not be supported in the official releases. Builds from the devel
+branch can have functionality changed and even removed at any time without
+compatibility support and without prior notice.
+
+**Also, we do not recommend any direct upgrades to 3.5 except from 3.4 to 3.5.**
+For example, upgrading from 3.3 to 3.5 is not recommended.
+
+git checkout v3.5.0 tag
+
+```bash
+git clone https://github.com/ceph/ceph-csi.git
+cd ./ceph-csi
+git checkout v3.5.0
+```
+
+**Note:** While upgrading please Ignore warning messages from kubectl output
+
+```console
+Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
+```
+
+To upgrade CephFS refer section [upgrade cephfs](#upgrading-cephfs) and
+to upgrade RBD refer section [upgrade rbd](#upgrading-rbd).
+
+With that, we have successfully upgraded cephfs csi from v3.4 to v3.5
+
 ## Upgrading from v3.3 to v3.4
```

Comment:
we don't need a separate section for 3.3 to 3.4 updates. just add a link how we are doing for 3.2 to 3.3 release https://github.com/ceph/ceph-csi/blob/9370d9d8bc777e4a59afcf8f544a053267f50ceb/docs/ceph-csi-upgrade.md#upgrading-from-v32-to-v33